## Q_1_Solution

### Consideration of hypothesis

Null Hypothesis(H0) = mean1 == mean2  (There is no significant difference in diameter of cutlet between two units)

Alternate Hypothesis(H1) = mean1 != mean2 (Significant difference exist in diameter of cutlet between two units)

### Types of Test

__Two sample, two tailed t test__

#Two tailed test is selected as we are eqauting mean of two samples

### Level of significance

LOS = 0.05

## t-test

In [2]:
import pandas as pd
import scipy 
import numpy as np
from scipy import stats

In [3]:
data = pd.read_csv("cutlets.csv")
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [4]:
stats.ttest_ind(data['Unit A'],data['Unit B'])

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

### Conclusion

p_value = 0.4722

As p_value > LOS(0.05)

__Null Hypothesis is NOT rejected__

__There is no significant difference in diameter of cutlet between two units__

## Q_2_Solution

### Consideration of hypothesis

Null Hypothesis(H0) = m1 == m2 == m3 == m4 (There is no significant difference in average Turn around time)

Alternate Hypothesis(H1) = mean are not same (Significant difference exist in Turn around time of the four labs)

### Types of Test

__one way ANOVA test__

#Due to presence of more than two samples one way ANOVA test is chosen

### Level of significance

LOS = 0.05

### One way anova test

In [6]:
import pandas as pd
import scipy.stats as stats
import numpy as np


In [7]:
lab = pd.read_csv("LabTAT.csv")
lab.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [8]:
stats.f_oneway(lab.iloc[:,0],lab.iloc[:,1],lab.iloc[:,2],lab.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

### Conclusion

p_value = 2.11e-57

As p_value < LOS(0.05)

__Null Hypothesis is rejected__

__Significant difference exist in Turn around time of the four labs__

## Q_3_Solution

### Consideration of hypothesis

Null Hypothesis(H0) = The variables are independent, there is no relation between male-female buyer ratio

Alternate Hypothesis(H1) = The variables are dependent, there is relation between male-female buyer ratio

### Types of Test

__Chi-square test__

#Chi square test is selected as more than one sample of multiple variables are present

### Level of significance

LOS = 0.05

### Chi-square test

In [9]:
import pandas as pd
import numpy as np
import scipy.stats as stat
from scipy.stats import chi2_contingency

In [10]:
data = pd.read_csv("BuyerRatio.csv")
data

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [11]:
data.drop(["Observed Values"],inplace = True,axis =1)    #inplace = True replaces old data with new changes
data

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [12]:
p_value = stat.chi2_contingency(data)[1]
print("P_value for the test is",np.round(p_value,3))

P_value for the test is 0.66


### Conclusion

p_value = 0.66

As p_value > LOS(0.05)

__Null Hypothesis is not rejected__

__The variables are independent, there is no relation between male-female buyer ratio__

## Q_4_Solution

### Consideration of hypothesis

Null Hypothesis(H0) = The variables are independent, the defective % doesn't vary by centre

Alternate Hypothesis(H1) = The variables are dependent,  the defective % varies by centre

### Types of Test

__Chi-square test__

#Chi square test is selected as more than one sample of multiple variables are present

### Level of significance

LOS = 0.05

### Chi-Square test

In [13]:
import pandas as pd
import numpy as np
import scipy.stats as stat
from scipy.stats import chi2_contingency

In [14]:
tele = pd.read_csv("Costomer+OrderForm.csv")
tele.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [15]:
print(tele["Phillippines"].value_counts(),
     tele["Indonesia"].value_counts(),
     tele["Malta"].value_counts(),
     tele["India"].value_counts())

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 Error Free    269
Defective      31
Name: Malta, dtype: int64 Error Free    280
Defective      20
Name: India, dtype: int64


In [16]:
array = np.array([[271,267,269,280],[29,33,31,20]])
array

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [17]:
p_value = stat.chi2_contingency(array)[1]
print("P_value for the test is",np.round(p_value,3))

P_value for the test is 0.277


### Conclusion

p_value = 0.277

As p_value > LOS(0.05)

__Null Hypothesis is not rejected__

__The variables are independent, the defective % doesn't vary by centre__